Apps/W1/HybridBaseDeployment/app/src/codeunits/cod4012.HandleCreateCompanyICFailure.al

```diff
@@ -7,7 +7,6 @@ codeunit 4012 "Handle Create Company Failure"
     begin
         ErrorMessage := GetLastErrorText();
         UpdateIntelligentCloudSetup(ErrorMessage);
-        Error(CompanyCreationFailedErr, ErrorMessage);
```

Comment:
@nikolakukrika 

I understand your concern.

We tried yesterday the cloud migration and got a permission error on the company creation. it is solved but the synch was not possible because the error state was set multiple times afterwards because of the retires of the scheduled tasks.

The problem is, that the Task Scheduler retries a failing codeunit up to 100 times even if it is the failure Codeunit.
You can look into the Session Event table where you see for example: Scheduled task failure codeunit failed. Task ID: dbe8d674-52d0-454b-8a77-3cb22039e976. Codeunit: 4012. Attempt: 101. Error: Dialog. 
The Main Codeunit can fail with an error, this is not problem. Problem is here the failure Codeunit, this should not end everytime in an error.

Please also consider this:
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-task-scheduler#general-flow
Quote:
"If the failure codeunit doesn't handle the error or fails itself, then the exception path is run to retry the failure codeunit. This retry flow continues in the same session until the task succeeds or until the maximum number of retries is exceeded. The session is then terminated."

You can also look into another Failure Codeunit of this app where no error is present: 40019 "Handle Fix Data Failure"

Our problem was a missing permission to create a new company.

I hope I could show you my concern. Thank you.

Reply:
Thanks a lot for the detailed description and for the fix. I will port the changes and test them before making and rolling out the HF.